mixer/pkg/api/grpcServer.go

```diff
@@ -142,6 +143,20 @@ func (s *grpcServer) check(ctx context.Context, req *mixerpb.CheckRequest,
 
 	lg.Debug("Dispatching to main adapters after running processors")
 	lg.Debuga("Attribute Bag: \n", checkBag)
+
+	if !s.dispatcher.HasCheckDestinations(ctx, checkBag) {
+		protoBag.ClearReferencedAttributes()
+		protoBag.Get("destination.uid") // should always be set; make it referenced for cache-key construction
```

Comment:
Why is this not happening by default on server-side checks? I'm confused. Without any rules, we should only use destination uid and context reporter as referenced attributes.

Reply:
Is there some bug that works against us here? In my local testing, Mixer would need to read those two attributes to scope config, and that's it.

Reply:
kubernetesenv adapter ends up pulling in `source.uid` and `source.ip` (and some others). this reduces that set down to the minimum.

Reply:
OK, that sounds like a bug. We should not be automatically making them referenced, unless a match condition or an instance uses the generated attribute.

Reply:
@kyessenov perhaps. that gets a bit trickier to track. this PR was a simple workaround.